src/transformers/generation_tf_utils.py

```diff
@@ -2624,7 +2625,7 @@ def sample_body_fn(generated, finished_sequences, cur_len, model_kwargs):
             if seed is not None:
                 sample_seed = seed
             else:
-                sample_seed = tf.cast(self.seed_generator.make_seeds(count=1)[:, 0], dtype=tf.int32)
+                sample_seed = tf.experimental.numpy.random.randint(tf.int32.min, tf.int32.max, (2,), dtype=tf.int32)
```

Comment:
Do we really have to rely on tf.experimental here? (e.g. it's not possible to convert this from a numpy array?)

Reply:
We can convert from a numpy array, but the execution speed penalty with XLA is huge (`t5-small` went from ~23ms/generation to ~56ms/generation on my machine). Sadly this is the only TF-based alternative to `tf.random.Generator` strategies

`tf.experimental.numpy` has been pretty stable since TF 2.4 🤔 

@Rocketknight1 what are your thoughts here?

Reply:
`tf.experimental.numpy` is extremely non-experimental by now, it's been around since TF 2.4 and is very unlikely to change significantly! I'm totally fine with using it across the codebase.